wordpress-mobile/WordPress-Android
# [Dependency Updates] Update `coilComposeVersion` to 2.2.2

Parent #17563
Batch Branch: [deps/main-batch-androidx-compose-kotlin](https://github.com/wordpress-mobile/WordPress-Android/tree/deps/main-batch-androidx-compose-kotlin)

This PR updates `coilComposeVersion` to [2.2.2](https://github.com/coil-kt/coil/releases/tag/2.2.2).

-----

Note that the latest [2.3.0](https://github.com/coil-kt/coil/releases/tag/2.3.0) update of `coilComposeVersion` is transitively updating `Kotlin` to [1.8.10](https://github.com/JetBrains/kotlin/releases/tag/v1.8.10) and as such will be done at a later point of time, after the `Compose Complier` is updated to target this newer `Kotlin` version.

-----

PS: @ovitrif I added you as the main reviewer, but not so randomly ([context](https://github.com/wordpress-mobile/WordPress-Android/pull/18303#issuecomment-1519988237)), since I just wanted someone from the WordPress team to be aware of and sign-off on that change for WPAndroid. I also added the @wordpress-mobile/apps-infrastructure team, but this in done only for monitoring purposes, as such, I am not expecting any active review from that team. Thus, feel free to merge this PR if you deem so.

-----

Fix List:

1. [Remove unsupported placeholders and errors from coil's async image](https://github.com/wordpress-mobile/WordPress-Android/commit/eecdefcebbb3da7bbc088bae01972af1b8c68ba8)

FYI: @ovitrif please take a close look at this commit's description and let me know your thoughts. I tried finding a solution to overcome this limitation but couldn't (I timeboxed my efforts). I then decided to remove both, the `placeholder` and `error`, to progress with this update and give you a heads-up on it in case you are able to think of a way forward. Below are a couple of my suggestion to move forward (post merging this PR):
- We ask help from design to draw us a `VectorDrawable` alternative of `bg_rectangle_placeholder_globe` and when that is done, we'll create another PR to add this functionality back before merging the [parent](https://github.com/wordpress-mobile/WordPress-Android/tree/deps/main-batch-androidx-compose-kotlin) branch to `trunk`, or 🎨
- We ignore this functionality altogether (for the time being), with the premise that it is just a nice to have but wouldn't break our flows, the UI would just show an empty image when the image can't be fetched from the API. 🤔 

PS: The `AsyncImage` official documentation can be found here: https://coil-kt.github.io/coil/compose/#asyncimage

Let me know your thoughts? 🙏 

Refactor List:

1. [Optimize imports on user avatar image](https://github.com/wordpress-mobile/WordPress-Android/commit/98c38791699848f1f14a509e8d4fcbf5f4d5035b)

-----

## To test:

1. See the dependency tree diff result and verify correctness.
2. Thoroughly smoke test any `Coil` enhanced `Compose` related screen, on both, the WordPress and Jetpack apps, and see if everything is working as expected.
3. In addition to the above smoke test, you can expand the below and follow the inner and more explicitly test steps within:

<details>
    <summary>1. Jetpack Migration Flow [JetpackMigrationFragment.kt + SiteList.kt + UserAvatarImage.kt]</summary>

ℹ️ This test applies to the `Jetpack` app.

- Install both apps.
- Clear cache/data of the `Jetpack` app and restart it.
- The migration flow should appear, verify the list of sites it is shown, including the user avatar
and any site image, and that everything is functioning as expected.

</details>

<details>
    <summary>2. Blaze Screen [BlazeOverlayFragment.kt]</summary>

ℹ️ This test applies to the `Jetpack` app.

- Go to `Posts` (or `Pages`) screen.
- Find a couple of posts from the list of posts, one with and without a featured image,
click on the `More` options and then on its `Promote with Blaze` option.
- Verify that the `Blaze` screen is shown, that it includes the post's (or page's) details,
with or without a featured image (depending on the post you selected), that the details
of this post (or page) are above the `Blaze this Post (or Page)` button, and that everything is
functioning as expected.

</details>


-----

## Regression Notes

1. Potential unintended areas of impact

    - Potential breakage or misbehaviour on any or all `Coil` enhanced `Compose` related screens, like the `Jetpack Migration Flow` screen and the `Blaze` green.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

PR submission checklist:


-----

UI Changes testing checklist:
